### ISSUE TITLE
Tracking December 2022 FOQUS Machine Learning Plugin Improvements

### ISSUE THREAD

**Issue by bpaul4** [CCSI-Toolset/FOQUS#1050]
This issue exists to track planned, ongoing and integrated features as part of milestone EY22.4.1.J - Improvements to the ML/AI interface in FOQUS. Each improvement is assumed to encompass documentation and example expansion for new features, and is marked with the planned release:

- [x] Investigate support for PyTorch neural network models (December 2022) #1081 
- [x] Investigate support for SciKit-Learn neural network models (December 2022) #1094 
- [x] Investigate support for Surrogate Modeling ToolBox neural networks (December 2022) (Not feasible due to derivative data requirement, see discussion below)

**Comment by bpaul4** [comment]
Including the pull requests linked above, the ML AI Plugin supports neural networks in TensorFlow Keras, PyTorch and Sci-kit Learn.

Surrogate Modeling Toolbox supports GENN (Gradient Enhanced Neural Network) models which require known derivative data that in practice would not be available to FOQUS users for their datasets. For me, this presented a high barrier to model generation during development testing. SMT also supports simpler surrogate types which are covered by ALAMO and IDAES surrogate methods, both of which are readily available via more accessible tools in FOQUS.

Seaborn was briefly considered, but does not support the types of machine learning models that could be loaded and run in FOQUS.

Considering everything above, I would consider expanded support for TensorFlow Keras, support for PyTorch, support for Sci-kit Learn, a large increase in test coverage for the FOQUS node, and improvements to the general plugin search methods to complete the milestone.
